auslib/test/test_db.py

```diff
@@ -381,17 +382,26 @@ def testUpdateClosesConnectionOnImplicitTransaction(self):
 
     def testUpdateWithChangeCallback(self):
         shared = []
-        self.test.onUpdate = lambda *x: shared.extend(x)
+
+        def onUpdate(*args, **kwargs):
+            shared.extend(args)
+            for _, v in kwargs.items():
+                shared.append(v)
+
+        self.test.onUpdate = onUpdate
         where = [self.test.id == 1]
-        what = dict(foo=123)
+        what = dict(foo=123, bar=42)
         self.test.update(changed_by='bob', where=where, what=what, old_data_version=1)
         # update adds data_version and id to the query, so we need to add that before comparing
         what["data_version"] = 2
         what["id"] = 1
         self.assertEquals(shared[0], self.test)
         self.assertEquals(shared[1], "UPDATE")
         self.assertEquals(shared[2], "bob")
+        what.pop('bar')
```

Comment:
It's better to avoid deleting things from `what` - the more it changes (especially through a delete), the harder it is to reason about it. Use a different variable, or a literal version of the data structure in the assert below.